src/rtree.c

```diff
@@ -158,6 +158,135 @@ rtree_child_read_hard(tsdn_t *tsdn, rtree_t *rtree, rtree_elm_t *elm,
 	return rtree_node_init(tsdn, rtree, level+1, &elm->child);
 }
 
+rtree_elm_t *
+rtree_elm_lookup_hard(tsdn_t *tsdn, rtree_t *rtree, rtree_ctx_t *rtree_ctx,
+    uintptr_t key, bool dependent, bool init_missing) {
+	uintptr_t leafkey = rtree_leafkey(rtree, key);
+	if (likely(key != 0)) {
+		/*
+		 * Search the remaining cache elements, and on success move the
+		 * matching element to the front.
+		 */
+		for (unsigned i = 1; i < RTREE_CTX_NCACHE; i++) {
```

Comment:
I'd still vote for keeping this cache searching part in the inlined function. The impact should be quite minimal but I'm not confident at all about application locality fitting well in only one leaf node.